packages/common-all/src/store/IConfigStore.ts

```diff
@@ -0,0 +1,59 @@
+import { ResultAsync } from "neverthrow";
+import { IDendronError } from "../error";
+import { DendronConfig, DendronConfigValue } from "../types";
+import { DeepPartial } from "../utils";
+
+export type ConfigValue = string | number | object;
+
+export type ConfigReadMode = "override" | "default";
+export type ConfigReadOpts = {
+  mode: ConfigReadMode;
+  useCache?: boolean;
```

Comment:
When would a caller not want to use a cache? Can this be safely removed from the read() params?

Reply:
The caching mechanism right now for config isn't too complex. Nothing is done when there is a miss. This is pretty much how it is implemented in `DConfig`, so for the initial phase I sticked with the way we are already doing.

I can change it so that we always read from cache if possible, but that opens up a possibility for some false negatives

Reply:
The way I was thinking about it is that whether you cache or not is an implementation detail of config store that a caller shouldn't need to worry about - it should just expect to get the right value regardless of whether a cache is used or not.

Reply:
that makes sense. I'll see what I can do here.

Reply:
Decided to remove caching for now. caching wasn't set up rigidly in the `DConfig` implementation, and was not used in most cases anyways.

Don't feel great having a half-baked caching mechanism be part of the config store at the moment, and there are some edge cases that need to be taken care of because of the overrides. Not worth spending time trying to make override work with caches at this moment.